plugins/powerd/fu-plugin-powerd.c

```diff
@@ -58,71 +95,9 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 			     g_dbus_proxy_get_name (data->proxy));
 		return FALSE;
 	}
-	return TRUE;
-}
-
-gboolean
-fu_plugin_update_prepare (FuPlugin *plugin,
-			  FwupdInstallFlags flags,
-			  FuDevice *device,
-			  GError **error)
-{
-	FuContext *ctx = fu_plugin_get_context (plugin);
-	FuPluginData *data = fu_plugin_get_data (plugin);
-	guint32 power_type = 0;
-	guint32 current_state = 0;
-	gdouble current_level = 0;
-	g_autoptr(GVariant) powerd_response = NULL;
-
-	/* permit updates when the device does not care about power conditions */
-	if (flags & FWUPD_INSTALL_FLAG_IGNORE_POWER)
-		return TRUE;
-
-	/* making method call to "GetBatteryState" through the proxy */
-	powerd_response = g_dbus_proxy_call_sync (data->proxy,
-						  "GetBatteryState",
-						  NULL,
-						  G_DBUS_CALL_FLAGS_NONE,
-						  -1,
-						  NULL,
-						  error);
-	if (powerd_response == NULL) {
-		g_prefix_error (error, "battery information was not loaded: ");
-		return FALSE;
-	}
 
-	/* parse and use for battery-check conditions */
-	g_variant_get (powerd_response,
-		       "(uud)",
-		       &power_type,
-		       &current_state,
-		       &current_level);
-
-	/* checking if percentage is invalid */
-	if (current_level < 1 || current_level > 100)
-		current_level = FU_BATTERY_VALUE_INVALID;
+	/* start a timer to repeatedly run fu_plugin_powerd_callback at a set interval */
+	g_timeout_add(5000, fu_plugin_powerd_callback, plugin);
```

Comment:
Also, `fu_plugin_powerd_refresh_cb` might be a more usual name for a callback

Reply:
Also, use G_CALLBACK(fu_plugin_powerd_refresh_cb) and then you don't have to do the gpointer cast above.